File: share_utils/ios/Classes/ShareUtilsPlugin.m
#import "ShareUtilsPlugin.h"

@implementation ShareUtilsPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"achjqz.com/share_utils"
            binaryMessenger:[registrar messenger]];
  ShareUtilsPlugin* instance = [[ShareUtilsPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getDeviceName" isEqualToString:call.method]) {
    result([[UIDevice currentDevice] name]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: share_utils/macos/Classes/ShareutilsPlugin.swift
import Cocoa
import FlutterMacOS

public class ShareUtilsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "achjqz.com/share_utils", binaryMessenger: registrar.messenger)
    let instance = ShareUtilsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
